lib/src/model/mi_scale_measurement.dart

```diff
@@ -134,24 +143,40 @@ class MiScaleMeasurement {
         weight: previousMeasurement.weight,
         stage: MiScaleMeasurementStage.MEASURED,
         unit: scaleData.unit,
+        impedance: null,
       );
     }
 
     // Finalize measurement if we are done measuring
     if (previousMeasurement.stage == MiScaleMeasurementStage.STABILIZED &&
         scaleData.measurementComplete) {
+      final impedance = scaleData.impedance;
       return MiScaleMeasurement(
         id: previousMeasurement.id,
         weight: previousMeasurement.weight,
         stage: MiScaleMeasurementStage.MEASURED,
         unit: scaleData.unit,
+        impedance: impedance,
       );
     }
 
     // Otherwise just return the previous measurement
     return previousMeasurement;
   }
 
+  /// The extra data that will be used to measure bmi, bone mass, fat percentage,... based on
+  MiScaleExtraData? getExtraData(MiScaleGender gender, int age, double height) {
+    final impedance = this.impedance;
+    if (impedance == null) return null;
+    return MiScaleExtraData(
+      gender: gender,
+      age: age,
+      height: height,
+      weight: weight,
+      impedance: impedance,
+    );
+  }
+
```

Comment:
I think we should probably add this to `MiScaleData` as well, as this would allow people who use `readScaleData` directly to access these values too.

Reply:
We can do that. But I think it should be optional. So if you only have the impedance you could also use this function to get all the values. (For example my app only saves the impedance value, and the BodyData is calculated at runtime)

Reply:
Agreed, this function can just be added to `MiScaleData` just like it is here to `MiScaleMeasurement`, right?

Reply:
Yes indeed